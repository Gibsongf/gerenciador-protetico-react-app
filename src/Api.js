export async function apiLogin(loginData) {
    const url =
        "http://https://gerenciador-protetico.adaptable.app/users/login";
    // console.log(loginData);
    try {
        const response = await fetch(url, {
            method: "post",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(loginData),
        });

        if (response.status === 200) {
            const data = await response.json();
            // console.log("Login successfully");
            localStorage.setItem("token", data.token);
            return true;
        } else {
            throw new Error();
        }
    } catch (error) {
        throw Error(error);
    }
}
export async function apiRegister(loginData) {
    const url =
        "http://https://gerenciador-protetico.adaptable.app/users/register";
    // console.log(loginData);
    try {
        const response = await fetch(url, {
            method: "post",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(loginData),
        });

        if (response.status === 200) {
            const data = await response.json();
            // console.log("Login successfully");
            localStorage.setItem("token", data.token);
            return true;
        } else {
            throw new Error();
        }
    } catch (error) {
        throw Error(error);
    }
}
async function setupFetch(url, reqMethod = "get", body) {
    if (!localStorage["token"]) {
        await apiLogin();
        // localStorage.setItem("token", data.token);
    }

    const reqConfig = {
        method: reqMethod,
        headers: {
            "Content-Type": "application/json",
            Authorization: "Bearer " + localStorage["token"],
        },
    };
    if (body) {
        reqConfig.body = JSON.stringify(body);
    }
    try {
        const response = await fetch(url, reqConfig);
        if (response.status === 200 || response.status === 400) {
            const data = await response.json();
            return data;
        }

        if (response.status === 401) {
            localStorage.removeItem("token");
            return "Old token";
        }
    } catch (err) {
        throw Error(err);
    }
}
export async function APItodos(category) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/${category}/todos`;
    const data = await setupFetch(url, "get");
    // console.log(data);
    return data;
}
export async function APIDetails(category, id) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/${category}/${id}`;
    const data = await setupFetch(url, "get");
    return data;
}
export async function APIGetServiceBy(id, by) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/servico/todos/${by}/${id}`;
    const data = await setupFetch(url, "get");
    return data;
}

export async function APIPostNewData(formData) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/${formData["category"]}/novo`;
    // console.log(formData);
    const data = await setupFetch(url, "post", formData);
    // console.log(data);
    return data;
}
export async function APIPutData(formData, id) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/${formData["category"]}/${id}/edit`;
    // console.log(formData);
    const data = await setupFetch(url, "put", formData);
    // console.log(data);

    return data;
}
export async function ApiOneExcel(id) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/export/${id}`;
    const reqConfig = {
        method: "Get",
        headers: {
            Authorization: "Bearer " + localStorage["token"],
        },
    };

    try {
        const response = await fetch(url, reqConfig);
        if (response.status === 200) {
            const blob = await response.blob();
            // console.log(data);
            return blob;
        }
    } catch (err) {
        throw Error(err);
    }
}

export async function ApiMonthExcel(id, date) {
    const url = `http://https://gerenciador-protetico.adaptable.app/api/export/${id}/${date}`;
    const reqConfig = {
        method: "Get",
        headers: {
            Authorization: "Bearer " + localStorage["token"],
        },
    };

    try {
        const response = await fetch(url, reqConfig);
        if (response.status === 200) {
            const blob = await response.blob();
            // console.log(data);
            return blob;
        }
    } catch (err) {
        throw Error(err);
    }
}
