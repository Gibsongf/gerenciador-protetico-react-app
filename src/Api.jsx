export async function apiLogin(loginData) {
    const url = "http://localhost:3000/users/login";
    // console.log(loginData);
    try {
        const response = await fetch(url, {
            method: "post",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(loginData),
        });

        if (response.status === 200) {
            const data = await response.json();
            console.log("Login successfully");
            localStorage.setItem("token", data.token);
            return true;
        } else {
            throw new Error();
        }
    } catch (error) {
        throw Error(error);
    }
}

async function setupFetch(url, reqMethod = "get", body) {
    if (!localStorage["token"]) {
        await apiLogin();
        // localStorage.setItem("token", data.token);
    }

    const reqConfig = {
        method: reqMethod,
        headers: {
            "Content-Type": "application/json",
            Authorization: "Bearer " + localStorage["token"],
        },
    };
    if (body) {
        reqConfig.body = JSON.stringify(body);
    }
    try {
        const response = await fetch(url, reqConfig);
        if (response.status === 200 || response.status === 400) {
            const data = await response.json();
            return data;
        }

        if (response.status === 401) {
            localStorage.removeItem("token");
            return "Old token";
        }
    } catch (err) {
        throw Error(err);
    }
}
export async function APItodos(category) {
    const url = `http://localhost:3000/api/${category}/todos`;
    const data = await setupFetch(url, "get");
    // console.log(data);
    return data;
}
export async function APIDetails(category, id) {
    const url = `http://localhost:3000/api/${category}/${id}`;
    const data = await setupFetch(url, "get");
    return data;
}
export async function APIGetServiceBy(id, by) {
    const url = `http://localhost:3000/api/servico/todos/${by}/${id}`;
    const data = await setupFetch(url, "get");
    return data;
}

export async function APIPostNewData(formData) {
    const url = `http://localhost:3000/api/${formData["category"]}/novo`;
    console.log(formData);
    const data = await setupFetch(url, "post", formData);
    // console.log(data);
    return data;
}
export async function APIPutData(formData, id) {
    const url = `http://localhost:3000/api/${formData["category"]}/${id}/edit`;
    // console.log(formData);
    const data = await setupFetch(url, "put", formData);
    // console.log(data);

    return data;
}
export async function ExcelLink(id) {
    const url = `http://localhost:3000/api/export/${id}`;
    const reqConfig = {
        method: "Get",
        headers: {
            Authorization: "Bearer " + localStorage["token"],
        },
    };

    try {
        const response = await fetch(url, reqConfig);
        if (response.status === 200) {
            const blob = await response.blob();
            // console.log(data);
            return blob;
        }
    } catch (err) {
        throw Error(err);
    }
}
