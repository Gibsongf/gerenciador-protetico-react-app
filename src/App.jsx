// import { useState } from 'react'
import { Header } from "./components/Header";

import { DentistasList } from "./pages/todos/DentistasList";
import { LocalList } from "./pages/todos/LocalList";
import { ProductList } from "./pages/todos/ProductList";
import { ServiçoList } from "./pages/todos/ServiçoList";
import { createContext } from "react";
import { Route, Router, Routes } from "react-router-dom";
import "./styles/App.css";
import { FormDentist } from "./components/Form/FormDentist";
import { FormLocal } from "./components/Form/FormLocal";
import { FormService } from "./components/Form/FormService";
import { FormProduct } from "./components/Form/FormProduct";
import { FormLogin } from "./components/Form/FormLogin";
import { DentistaDetails } from "./pages/detalhes/DentistaDetails";
import { LocalDetails } from "./pages/detalhes/LocalDetails";
import { ServiçoDetails } from "./pages/detalhes/ServiçoDetails";
import { ProdutoDetails } from "./pages/detalhes/ProdutoDetails";

export const AppContext = createContext({
    errorMsg: {},
    saveDbId: () => {},
});
function App() {
    const token = localStorage.getItem("token");

    // If has token and is valid send to index else delete token and redirect login
    if (!token) {
        return <FormLogin />;
    }
    const saveDbId = (e) => {
        localStorage.setItem("dbID", e.target.id);
    };
    // Need to change phone format of saving and read todo of details
    Router;
    return (
        <div className="App">
            <AppContext.Provider value={{ errorMsg: {}, saveDbId }}>
                <Header />
                <Routes>
                    <Route path="/dentista">
                        <Route
                            path="/dentista/todos"
                            element={<DentistasList />}
                        />
                        <Route
                            path="/dentista/:id"
                            element={<DentistaDetails />}
                        />
                    </Route>
                    <Route path="/todos-locais" element={<LocalList />} />
                    <Route path="/todos-servicos" element={<ServiçoList />} />
                    <Route path="/todos-produtos" element={<ProductList />} />
                </Routes>
                {/* All List  */}
                {/* <LocalList /> */}
                {/* <DentistasList />*/}
                {/* <LocalList /> */}
                {/* <ProductList /> */}
                {/* <ServiçoList /> */}

                {/* Forms */}
                {/* <FormLocal /> */}
                <FormDentist />
                {/* <FormProduct /> */}
                {/* <FormService /> */}

                {/* Details */}
                {/* <DentistaDetails /> */}
                {/* <LocalDetails/> */}
                {/* <ServiçoDetails/> */}
                {/* <ProdutoDetails/> */}
            </AppContext.Provider>
        </div>
    );
}

export default App;
