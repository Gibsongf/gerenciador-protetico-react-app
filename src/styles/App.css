#root {
  /* max-width: 1280px; */
  /* margin: 0 auto; */
  /* padding: 2rem; */
  /* text-align: center; */
}

.App {
  display: grid;
  gap: 25px;
}

.todos-table {
  /* width: -webkit-fill-available; */
  /* margin: 0 auto; */
  padding: 2rem;
  border-collapse: collapse;


}

.todos-table,
td,
th {
  border: 3px solid black;
}

/* th {
  padding-left: 10px;
  padding-right: 10px;
} */

td {
  padding: 10px;
}

.nav {
  border: 1px solid black;
  display: flex;
  justify-content: space-around;
}



.login-form {
  border: 2px solid black;
  border-radius: 10px;
  padding: 10px;
}

.username,
.password {
  display: flex;
  flex-direction: column;
}

/* .logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
} */