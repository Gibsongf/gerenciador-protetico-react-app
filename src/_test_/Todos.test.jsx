/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
// import { FormLogin } from "../components/Form/FormLogin";
import { expect, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import { HashRouter } from "react-router-dom";
import App from "../App";
import { apiLogin } from "../Api";

// vi.mock("../Api", () => ({
//     apiLogin: jest.fn().mockReturnValue(false),
// }));
vi.mock("../Api", () => {
    return {
        default: { apiLogin: vi.fn() },
        apiLogin: vi.fn(),
        // etc...
    };
});
describe("App component", () => {
    it("renders form correct when no token registered", () => {
        const { container } = render(
            //has to use browserRouter for useNavigate to work
            <HashRouter>
                <App />
            </HashRouter>
        );
        const form = screen.getByRole("login-form");
        expect(container).toMatchSnapshot();
        expect(form).toBeInTheDocument();
    });
    it.only("Form Login successfully", async () => {
        const user = userEvent.setup();
        localStorage.setItem("token", "");
        render(
            //has to use browserRouter for useNavigate to work
            <HashRouter>
                <App />
            </HashRouter>
        );

        const username = screen.getByRole("textbox", {
            name: "Nome de Usu√°rio:",
        });
        const password = screen.getByLabelText("Senha:");
        const button = screen.getByRole("button", { name: "Entrar" });
        const form = screen.getByRole("login-form");
        await user.tripleClick(username);
        await user.keyboard("test");
        await user.tripleClick(password);
        await user.keyboard("test");
        await user.tripleClick(button);
        expect(username.value).toBe("test");
        expect(password.value).toBe("test");
    });
});
