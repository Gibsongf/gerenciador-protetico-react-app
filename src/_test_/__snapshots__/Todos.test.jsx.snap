// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > Form Login successfully 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="login-container"
    >
      <form
        action=""
        class="login-form"
        method="POST"
        role="login-form"
      >
        <div
          class="username"
        >
          <label
            for="username"
          >
            Nome de Usuário:
          </label>
          <input
            id="username"
            name="username"
            required=""
            type="text"
            value="test"
          />
        </div>
        <div
          class="password"
        >
          <label
            for="password"
          >
            Senha:
          </label>
          <input
            id="password"
            name="password"
            required=""
            type="password"
            value="test"
          />
        </div>
        <button
          class="login-enter"
          type="submit"
        >
          Entrar
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`App component > renders form correct when no token registered 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="login-container"
    >
      <form
        action=""
        class="login-form"
        method="POST"
        role="login-form"
      >
        <div
          class="username"
        >
          <label
            for="username"
          >
            Nome de Usuário:
          </label>
          <input
            id="username"
            name="username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="password"
        >
          <label
            for="password"
          >
            Senha:
          </label>
          <input
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
        <button
          class="login-enter"
          type="submit"
        >
          Entrar
        </button>
      </form>
    </div>
  </div>
</div>
`;
